lib/convert_list_to_deb822.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+"""
+This script is called after running do-release-upgrade in a machine.
+This converts list files to deb822 files when upgrading to Noble.
+"""
+
+import logging
+import os
+import sys
+
+from aptsources.sourceslist import SourceEntry  # type: ignore
+
+from uaclient import entitlements
+from uaclient.apt import _get_sources_file_content
+from uaclient.cli import setup_logging
+from uaclient.config import UAConfig
+from uaclient.system import (
+    ensure_file_absent,
+    get_release_info,
+    load_file,
+    write_file,
+)
+from uaclient.util import set_filename_extension
+
+if __name__ == "__main__":
+    series = get_release_info().series
+    if series != "noble":
+        sys.exit(0)
+
+    setup_logging(logging.DEBUG)
+    cfg = UAConfig()
+
+    for entitlement_class in entitlements.ENTITLEMENT_CLASSES:
+        if not issubclass(
+            entitlement_class, entitlements.repo.RepoEntitlement
+        ):
+            continue
+
+        entitlement = entitlement_class(cfg)
+
+        filename = set_filename_extension(entitlement.repo_file, "list")
```

Comment:
Just checking, but a normal `do-release-upgrade` will rename "third-party" `.list` files to something else, and I'm presuming it will not consider the entitlement repo_files as third party, correct? Because otherwise at this point these files will have been renamed to something like `.list.distUpgrade` or other.

Reply:
So when `do-release-upgrade` considers some of our services as third-parties (which it should not do, btw...) the `.list` file is commented out, and the `.list.distUpgrade` file is not a rename, but rather a backup.
As of now, the `update_lts_contract` script will handle this situation, by removing the file and recreating the apt config for the given entitlement - which will be a `.sources` file in noble.
For the future, we can improve this situation by making sure we are always recognized as valid sources.